src/Microsoft.Net.Http.Headers/DateTimeFormatter.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Net.Http.Headers
+{
+    internal static class DateTimeFormatter
+    {
+        private static readonly DateTimeFormatInfo FormatInfo = CultureInfo.InvariantCulture.DateTimeFormat;
+
+        // The format is "ddd, dd MMM yyyy HH:mm:ss GMT".
+        private const int Rfc1123DateLength = 29;
+
+        // ASCII numbers are in the range 48 - 57.
+        private const int AsciiNumberOffset = 0x30;
+
+        private const string Gmt = "GMT";
+        private const char Comma = ',';
+        private const char Space = ' ';
+        private const char Colon = ':';
+
+        public static unsafe string ToRfc1123String(this DateTimeOffset dateTime)
+        {
+            var offset = 0;
+            char* target = stackalloc char[Rfc1123DateLength];
+            var universalDateTime = dateTime.ToUniversalTime();
+
+            offset = FormatDayOfWeek(universalDateTime.DayOfWeek, target, offset);
+            offset = Format(Comma, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatNumber(universalDateTime.Day, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatMonth(universalDateTime.Month, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatYear(universalDateTime.Year, target, offset);
+            offset = Format(Space, target, offset);
+            offset = FormatTimeOfDay(universalDateTime.TimeOfDay, target, offset);
+            offset = Format(Space, target, offset);
+            offset = Format(Gmt, target, offset);
+
+            return new string(target, 0, offset);
+        }
+
+        private static unsafe int FormatDayOfWeek(DayOfWeek dayOfWeek, char* target, int offset)
+        {
+            return Format(FormatInfo.AbbreviatedDayNames[(int) dayOfWeek], target, offset);
```

Comment:
Yeah, using those regressed perf substantially:

![image](https://cloud.githubusercontent.com/assets/582487/18999637/f221a494-873e-11e6-811f-f010dfa975c6.png)

Caching the arrays *did* help, but it's still worse than it was:

![image](https://cloud.githubusercontent.com/assets/582487/18999623/dd2f8f1a-873e-11e6-8bb3-9768592b4c1a.png)